api/client/client.go

```diff
@@ -259,11 +324,28 @@ func (c *Client) Listen(storeID, modelName, entityID string) (<-chan ListenEvent
 				}
 				break
 			}
-			bytes := []byte(event.GetEntity())
-			channel <- ListenEvent{data: bytes}
+			var actionType ActionType
+			switch event.GetAction() {
+			case pb.ListenReply_CREATE:
+				actionType = ActionCreate
+			case pb.ListenReply_DELETE:
+				actionType = ActionDelete
+			case pb.ListenReply_SAVE:
+				actionType = ActionSave
+			default:
+				channel <- ListenEvent{err: fmt.Errorf("unknown listen reply action %v", event.GetAction())}
+				break L
+			}
+			action := Action{
+				Model:    event.GetModelName(),
+				Type:     actionType,
+				EntityID: event.GetEntityID(),
+				Entity:   event.GetEntity(),
```

Comment:
Sending the entity JSON bytes along in the action. This is about as good as we can do since we don't know what Go type the bytes represent (the user can register to listen for updates to many/different types of models).

Reply:
Sounds handy.